DependencyGraph/src/main/kotlin/org/jetbrains/research/ictl/fileimportance/DependencyExtractor.kt

```diff
@@ -0,0 +1,73 @@
+package org.jetbrains.research.ictl.fileimportance
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.JavaRecursiveElementVisitor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiManager
+import com.intellij.psi.PsiRecursiveElementVisitor
+import com.intellij.psi.search.FilenameIndex
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.refactoring.suggested.startOffset
+import java.nio.file.Path
+
+class DependencyExtractor(
+    val project: Project,
+    val extension: String,
+    val projectPath: Path
+) : PsiRecursiveElementVisitor() {
+    private val edgesBuffer: MutableList<DependencyEdge> = mutableListOf()
+    private val foundElementOffsets = mutableListOf<Int>()
```

Comment:
I am not sure I fully understand why is this required. @vharatian can you explain it please?

Reply:
Do you mean the abstraction? Since the dependency extractors for different languages were different before an abstraction was required for their API.

Reply:
Im ean the `foundElementOffsets`. What is it for?